src/PdfReader/useMeasure.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+
+declare const window: Window &
+  typeof globalThis & {
+    ResizeObserver: any;
+  };
+export type Dimensions = Pick<
+  DOMRectReadOnly,
+  'x' | 'y' | 'top' | 'left' | 'right' | 'bottom' | 'height' | 'width'
+>;
+export type UseMeasureRef<E extends Element = Element> = (element: E) => void;
+export type UseMeasureResult<E extends Element = Element> = [
+  UseMeasureRef<E> | null,
+  Dimensions | null
+];
+
+export default function useMeasure<
+  E extends Element = Element
+>(): UseMeasureResult {
+  // this is a little trick to get a reference to an HTML element. Using useRef wouldn't
+  // work because we actually need rerenders when it changes, to update the useLayoutEffect
+  const [element, ref] = React.useState<E | null>(null);
+  const [rect, setRect] = React.useState<Dimensions | null>(null);
+  const observer = React.useMemo(
+    () =>
+      new window.ResizeObserver(
+        (
+          entries: {
+            contentRect: {
+              x: any;
+              y: any;
+              width: any;
+              height: any;
+              top: any;
+              left: any;
+              bottom: any;
+              right: any;
+            };
+          }[]
+        ) => {
+          if (entries[0]) {
+            const {
+              x,
+              y,
+              width,
+              height,
+              top,
+              left,
+              bottom,
+              right,
+            } = entries[0].contentRect;
+            setRect({ x, y, width, height, top, left, bottom, right });
+          }
+        }
+      ),
+    []
+  );
+
+  React.useLayoutEffect(() => {
+    if (!element) return;
+    observer.observe(element);
+    return () => {
+      observer.disconnect();
+    };
+  }, [element, observer]);
+
+  return [ref, rect];
```

Comment:
There's also a typescript error here that cascades into a type error here: https://github.com/NYPL-Simplified/web-reader/pull/42/files#diff-8c9d1a11768b098f16811aec65d3a9bc7c4baad23d0f7c13fea2150d1cf74d2dR433

I'd love some thoughts on the type issues that happen here, because I'm not sure I understand all of the generics going on, and would love to know how one would debug this.